API/NoDocs/Aspire.Azure.Messaging.EventHubs/AzureMessagingEventHubsBaseSettings.cs

```diff
@@ -0,0 +1,17 @@
+// Assembly 'Aspire.Azure.Messaging.EventHubs'
+
+using System.Runtime.CompilerServices;
+using Aspire.Azure.Common;
+using Azure.Core;
+
+namespace Aspire.Azure.Messaging.EventHubs;
+
+public abstract class AzureMessagingEventHubsBaseSettings : IConnectionStringSettings
```

Comment:
Folks on my team are pretty allergic to the word `Base` and we're not using it in the EH client libraries today (see [Azure.Messaging.EventHubs](https://apiview.dev/Assemblies/Review/050a8e78e2e94b92b38ec4920307efda) and [Azure.Messaging.EventHubs.Processor](https://apiview.dev/Assemblies/Review/7c00b3c911484f67a433d30f8eb6fb06)).  Are we adding this base class because we think it'll help customers or because we're trying to make our implementation slightly easier?  I struggle to imagine people wanting to polymorph over an input only settings class.

Reply:
Inlining the properties seems fine.

Reply:
A couple thoughts:

1. My current plan is to keep EventHubs in "pre-release" even when the rest of .NET Aspire ships stable `8.0.0`. So we can tweak this API later. This component came in pretty late in preview5 and follows a bit of a different pattern than the others. So I think we need a little more bake time.
2. We use the base class internally:

https://github.com/dotnet/aspire/blob/e2ba459e1505c82219efef201d086481c1f59df2/src/Components/Aspire.Azure.Messaging.EventHubs/EventHubsComponent.cs#L16-L31

inlining the properties will just mean some more code duplication internally, but that isn't a huge deal.

> Are we adding this base class because we think it'll help customers or because we're trying to make our implementation slightly easier?

It's mostly the "don't repeat yourself" and making our implementation slightly easier.

> allergic to the word Base

Agreed we can drop `Base`.

____

In the end, I think we could make it work to not have base classes here, but it feels less "Object Oriented" to say "let's not abstract commonalitites in these classes".  Note that consumers are never expected to create these objects, they get passed the object in the `configureSettings` callback.